programs/helium-entity-manager/src/instructions/temp_pay_mobile_onboarding_fee_v0.rs

```diff
@@ -0,0 +1,110 @@
+use crate::{rewardable_entity_config_seeds, state::*, TESTING};
+use anchor_lang::prelude::*;
+use anchor_spl::{
+  associated_token::AssociatedToken,
+  token::{Mint, Token, TokenAccount},
+};
+use data_credits::{
+  cpi::{
+    accounts::{BurnCommonV0, BurnWithoutTrackingV0},
+    burn_without_tracking_v0,
+  },
+  program::DataCredits,
+  BurnWithoutTrackingArgsV0, DataCreditsV0,
+};
+use helium_sub_daos::{
+  cpi::{accounts::TrackDcOnboardingFeesV0, track_dc_onboarding_fees_v0},
+  program::HeliumSubDaos,
+  DaoV0, SubDaoV0, TrackDcOnboardingFeesArgsV0,
+};
+
+#[derive(Accounts)]
+pub struct TempPayMobileOnboardingFeeV0<'info> {
+  #[account(mut)]
+  pub dc_fee_payer: Signer<'info>,
+  #[account(
+    mut,
+    associated_token::mint = dc_mint,
+    associated_token::authority = dc_fee_payer,
+  )]
+  pub dc_burner: Box<Account<'info, TokenAccount>>,
+  #[account(
+    has_one = sub_dao,
+    constraint = rewardable_entity_config.settings.is_mobile() && (rewardable_entity_config.symbol == "MOBILE" || TESTING),
+  )]
+  pub rewardable_entity_config: Box<Account<'info, RewardableEntityConfigV0>>,
+  #[account(mut,
+has_one = dao)]
+  pub sub_dao: Box<Account<'info, SubDaoV0>>,
+  #[account(
+    has_one = dc_mint,
+  )]
+  pub dao: Box<Account<'info, DaoV0>>,
+  #[account(mut)]
+  pub dc_mint: Box<Account<'info, Mint>>,
+  #[account(
+    seeds = [
+      "dc".as_bytes(),
+      dc_mint.key().as_ref()
+    ],
+    seeds::program = data_credits_program.key(),
+    bump = dc.data_credits_bump,
+    has_one = dc_mint
+  )]
+  pub dc: Account<'info, DataCreditsV0>,
+  #[account(mut)]
+  pub mobile_info: Box<Account<'info, MobileHotspotInfoV0>>,
```

Comment:
Missing a seeds check to ensure that this is part of the correct rewardable entity config. Otherwise you can just make a new dao with a madeup data credits and "pay" on a real helium hotspot